features/dd-sdk-android-flags/api/apiSurface

```diff
@@ -5,15 +5,14 @@ data class com.datadog.android.flags.FlagsConfiguration
   class Builder
     fun setEnableExposureLogging(Boolean): Builder
     fun build(): FlagsConfiguration
-object com.datadog.android.flags.featureflags.FlagsClient
-  fun isRegistered(com.datadog.android.api.SdkCore = Datadog.getInstance()): Boolean
-  fun get(com.datadog.android.api.SdkCore = Datadog.getInstance()): FlagsProvider
-interface com.datadog.android.flags.featureflags.FlagsProvider
+interface com.datadog.android.flags.featureflags.FlagsClient
   fun setContext(com.datadog.android.flags.featureflags.model.EvaluationContext)
   fun resolveBooleanValue(String, Boolean): Boolean
   fun resolveStringValue(String, String): String
   fun resolveNumberValue(String, Number): Number
   fun resolveIntValue(String, Int): Int
   fun resolveStructureValue(String, org.json.JSONObject): org.json.JSONObject
+  companion object 
+    fun instance(com.datadog.android.api.SdkCore = Datadog.getInstance()): FlagsClient
```

Comment:
to be consistent with existing API, it is better to call it `get`, like we have here https://github.com/DataDog/dd-sdk-android/blob/6de04ce550bd7364b62e51dbeccc4ac8afac90b7/features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/GlobalRumMonitor.kt#L57